src/main/play-doc/general/MigrationGuide.md

```diff
@@ -33,18 +33,325 @@ In additional to disabling Elasticsearch via configuration, a service lookup mus
+
+This behaviour can now be declared with a request ACL.
+
+###### Migrating build.sbt
+
+Modify the `Endpoint` declared within the `BundleKeys.endpoints`.
+
+From:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, Set(URI("http://my-service?preservePath")))
+)
+```
+
+To:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, "my-service",
+    RequestAcl(Http("^/my-service".r))
+  )
+)
+```
+
+###### Migrating bundle.conf
+
+Modify the `endpoints` section declared within the `bundle.conf` as such.
+
+From:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        services      = ["http://my-service?preservePath"]
+      }
+    }
+```
+
+To:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        service-name  = "my-service"
+        acls          = [
+          {
+            http = {
+              requests = [
+                {
+                  path-beg = "/my-service"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+```
+
+#### Supplying customized HAProxy template configuration
+
+The following services definition will require a customized HAProxy configuration.
+
+The [[Dynamic proxy configuration|DynamicProxyConfiguration]] describes the process of generating a customized HAProxy configuration.
+
+Generating a customized HAProxy configuration will rely on the [[ifAcl|DynamicProxyConfiguration#ifAcl]] directive which accepts `system`, `systemVersion`, and `endpointLabel` as input argument.
+
+When declaring bundle settings via `build.sbt`, unless declared explicitly, the `system` and `systemVersion` settings may fallback to a default value as described by [sbt-conductr settings](https://github.com/typesafehub/sbt-conductr#bundle-settings).
+
+
+##### Customized port number
+
+A service may declare an endpoint exposed on a custom port number. Given the service URI `http://:6789/my-service`, the service is accessible from the outside world on HTTP port `6789` under `/my-service` context.
+
+Given the following `build.sbt` entry:
+
+```
+name := "my-bundle"
+BundleKeys.compatibilityVersion := "1.1"
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, Set(URI("http://:6789/my-service")))
+)
+
+```
+
+Or the following `bundle.conf` entry:
+
+```
+    system             = "my-bundle"
+    systemVersion      = "1.1"
+    endpoints          = {
+      "endpoint-label" = {
+        bind-protocol  = "http"
+        bind-port      = 0
+        services       = ["http://:6789/my-service"]
+      }
+    }
+```
+
+The following customized template will provide the equivalent proxy mapping:
```

Comment:
For me it is unclear if the entry in the `build.sbt` / `bundle.conf` needs to be removed or not when creating a customized template?

If I still need to declare an endpoint in the `build.sbt` / `bundle.conf`, shouldn't I then use the new RequestACL format instead of using the "deprecated" services declaration. This should be documented here.